ts/src/config.ts

```diff
@@ -77,9 +78,25 @@ export interface Config extends AuthenticationConfig {
   // stack depth may increase overhead of profiling.
   heapMaxStackDepth?: number;
 
-  // Time to wait before trying to create a profile again if profile creation
-  // fails.
-  backoffMillis?: number;
+  // On first error during profile creation, if the backoff is not specified
+  // by the server response, then profiler will between 0 and
+  // initialBackoffMillis before asking the server to create a profile again.
+  // After a successful profile creation, the backoff will be reset to
+  // initialExpBackoffMillis.
+  initialBackoffMillis?: number;
+
+  // If the backoff is not specified by the server response, then profiler will
+  // wait at most maxBackoffMillis before asking server to create a profile
+  // again.
+  maxBackoffMillis?: number;
+
+  // On each consecutive error in profile creation, the maximum backoff will
+  // increase by this factor. The backoff will be random value selected
+  // from a uniform distribution between 0 and the maximum backoff.
+  backoffMultiplier?: number;
+
+  // Server-specified backoffs will be capped at backoffLimitMillis.
+  backoffLimitMillis?: number;
```

Comment:
Document why it's capped. We don't cap the server instructed backoff for other agents.

I also find maxBackoffMillis and backoffLimitMillis names unnecessarily inconsistent and not super clear.
* You end up using three terms here: "max", "limit" and "cap" to mean the same thing. You also sometimes use "max" to mean the current value of the backoff before the randomization applies which may confuse the reader even more, see comment above.
* The names do not make it clear that the first thing is about capping the backoff on errors and the second thing is about capping the server-instructed backoff.

I would use the "cap" term and clarify the naming / terminology here and in the code (variable names).

Reply:
done.